app/routes/goal_routes.py

```diff
@@ -1 +1,134 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+from app.models.task import Task
+from app.models.goal import Goal
+from sqlalchemy import asc, desc
+from ..db import db
+import requests
+import json
+import os
+
+bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+
+@bp.post("")
+def create_goal():
+    request_body = request.get_json()
+    try:
+        new_goal = Goal.from_dict(request_body)
+    except KeyError as e:
+        response = {"details": "Invalid data"}
+        abort(make_response(response, 400))
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    response = {"goal": new_goal.to_dict()}
+    return response, 201
+
+
+@bp.get("")
+def get_all_goals():
+    query = db.select(Goal)
+    sort_param = request.args.get("sort")
+    if sort_param == "asc":
+        query = query.order_by(asc(Goal.title))
+    if sort_param == "desc":
+        query = query.order_by(desc(Goal.title))
+    else:
+        query = query.order_by(Goal.id)
+
+    goals = db.session.scalars(query)
+
+    tasks_response = []
+    for goal in goals:
+        tasks_response.append(goal.to_dict())
+
+    return tasks_response
+
+
+@bp.get("/<goal_id>")
+def get_one_task_by_id(goal_id):
+    goal = validate_goal(goal_id)
+
+    return {"goal": goal.to_dict()}
+
+# should it be refactored with from_dict??
```

Comment:
Are you asking if line 53 should invoke `from_dict` instead of `to_dict`?

We want to return a JSON response to our client when they make a GET request to `/goals/<goal_id>`. Therefore we need to send back a representation of the `Goal` object using a dictionary. 

Your `from_dict` method returns an instance of a class. Flask does not know how to translate a custom Python object into JSON, but it can automatically convert a dictionary (representing your object) into JSON